docs/design.md

```diff
@@ -1,96 +1,99 @@
-# Ansible Playbook Bundle (APB) Spec File
+# Design
+
+## Overview
+An [Ansible Playbook Bundle (APB)](https://github.com/fusor/ansible-playbook-bundle)
+borrows several concepts from the [Nulecule](https://github.com/projectatomic/nulecule)
+or [Atomicapp](http://www.projectatomic.io/docs/atomicapp/) project, namely the concept of a short
+lived container with the sole purpose of orchestrating the deployment of the intended application. For the case
+of APB, this short lived container is the APB; a container with an Ansible runtime environment
+plus any files required to assist in orchestration such as playbooks, roles, and extra dependencies.
+Specification of an APB is intended to be lightweight, consisting of several named playbooks and a
+metadata file to capture information such as parameters to pass into the application.
+
+## Workflow
+APB is broken up into the following steps.
+
+  1. [Preparation](#preparation)
+    * [APB init](#apb-initialization)
+    * [Spec File](#spec-file)
+    * [Actions](#actions) (provision, deprovision, bind, unbind)
+  1. [Build](#build)
+  1. [Deploy](#deploy)
+
+### Preparation
+
+#### APB Initialization
+![Prepare](images/apb-prepare.png)
+The first step to creating an APB is to run the `apb init` command, which will create the required skeleton directory structure, and a few required files (e.g. `apb.yml` spec file) for the APB.
+
+##### Directory Structure
+The following show an example directory structure of an APB.
+```bash
+example-apb/
+├── Dockerfile
+├── apb.yml
+└── roles/
+│   └── example-apb-openshift
+│       ├── defaults
+│       │   └── main.yml
+│       └── tasks
+│           └── main.yml
+└── playbooks/
+    └── provision.yml
+    └── deprovision.yml
+    └── bind.yml
+    └── unbind.yml
+```
 
-Example:
+#### Spec File
 
-```yaml
-name: fusor/etherpad-apb
-description: Note taking web application
-bindable: true
-async: optional
-parameters:
-  - name: hostport
-    description: The host TCP port as the external endpoint
-    type: int
-    default: 9001
-  - name: db_user
-    description: Database User
-    type: string
-  - name: db_pass
-    description: Database Password
-    type: string
-  - name: db_name
-    description: Database Name
-    type: string
-  - name: db_host
-    description: Database service hostname/ip
-    default: mariadb
-    type: string
-  - name: db_port
-    description: Database service port
-    type: int
-    default: 3306
-```
-Or with an empty parameters field
-```yaml
-name: example/no-parameters 
-description: Example showing no parameters
+The `apb init` will create an example APB Spec `apb.yml` File as shown below:
+```yml
+name: my-apb
+image: <docker-org>/my-apb
+description: "My New APB"
 bindable: false
```

Comment:
I would show a more full example with a comment about empty parameters.

                                                            
```yaml
 name: fusor/etherpad-apb
 description: Note taking web application
 bindable: true
 async: optional
 parameters:
   - name: hostport
     description: The host TCP port as the external endpoint
     type: int
     default: 9001
   - name: db_user
     description: Database User 
     type: string
```

For an APB that does not have any parameters, that field would look like:

```yaml
parameters: []
```